x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinator.java

```diff
@@ -253,6 +253,8 @@ void updateAutoFollowers(ClusterState followerClusterState) {
             return;
         }
 
+        this.patterns = Set.copyOf(autoFollowMetadata.getPatterns().keySet());
+
```

Comment:
I think we should set the patterns to the empty list above when `autoFollowMetadata == null` too (though I think we never go back to null, so mostly for completeness).

Reply:
I believe we are also not clearing/stopping `autoFollowers` once metadata is null. If this happens for whatever reason then it might result in auto-followers still running but not producing any stats

Reply:
Good point, perhaps we can simply add an assert before the return above to express the intent here?